pkg/infra/main.go

```diff
@@ -72,9 +72,15 @@ func gameLoop(globalState state.State, agentMap map[commons.ID]agent.Agent, game
 				fight.DealDamage(attack, agentMap, &globalState)
 			} else {
 				globalState.MonsterHealth = commons.SaturatingSub(globalState.MonsterHealth, attackSum)
-				if globalState.MonsterHealth > 0 {
+				if globalState.MonsterHealth > 0 && shieldSum < globalState.MonsterAttack {
 					damageTaken := globalState.MonsterAttack - shieldSum
 					fight.DealDamage(damageTaken, agentMap, &globalState)
+
+					logging.Log.WithFields(logging.LogField{
+						"Damage":        damageTaken,
+						"monsterAttack": globalState.MonsterAttack,
+						"ShieldSum":     shieldSum,
+					}).Info("Damage")
```

Comment:
Oops sorry this one slipped my first comments, but the same thing needs to be done here. (I can't quite format it properly on GitHub so maybe just do it locally)
```suggestion
					logging.Log(logging.Info, logging.LogField{
					    "Damage": damageTaken,
					    "monsterAttack": globalState.MonsterAttack,
					    "ShieldSum": shieldSum,
					}, "Damage")
```

Reply:
done i think